frontend/src/app/features/user-preferences/notifications-settings/page/notifications-settings-page.component.ts

```diff
@@ -201,24 +201,24 @@ export class NotificationsSettingsPageComponent extends UntilDestroyedMixin impl
           return;
         }
 
-        const projectSettings = new FormArray([]);
+        const projectSettings = new UntypedFormArray([]);
         projectSettings.clear();
         settings
           .sort(
             (a, b):number => a._links.project.title!.localeCompare(b._links.project.title!),
           )
-          .forEach((setting) => projectSettings.push(new FormGroup({
-            project: new FormControl(setting._links.project),
-            assignee: new FormControl(setting.assignee),
-            responsible: new FormControl(setting.responsible),
-            workPackageCreated: new FormControl(setting.workPackageCreated),
-            workPackageProcessed: new FormControl(setting.workPackageProcessed),
-            workPackageScheduled: new FormControl(setting.workPackageScheduled),
-            workPackagePrioritized: new FormControl(setting.workPackagePrioritized),
-            workPackageCommented: new FormControl(setting.workPackageCommented),
-            startDate: new FormControl(setting.startDate),
-            dueDate: new FormControl(setting.dueDate),
-            overdue: new FormControl(setting.overdue),
+          .forEach((setting) => projectSettings.push(new UntypedFormGroup({
+            project: new UntypedFormControl(setting._links.project),
+            assignee: new UntypedFormControl(setting.assignee),
+            responsible: new UntypedFormControl(setting.responsible),
+            workPackageCreated: new UntypedFormControl(setting.workPackageCreated),
+            workPackageProcessed: new UntypedFormControl(setting.workPackageProcessed),
+            workPackageScheduled: new UntypedFormControl(setting.workPackageScheduled),
+            workPackagePrioritized: new UntypedFormControl(setting.workPackagePrioritized),
+            workPackageCommented: new UntypedFormControl(setting.workPackageCommented),
+            startDate: new UntypedFormControl(setting.startDate),
+            dueDate: new UntypedFormControl(setting.dueDate),
+            overdue: new UntypedFormControl(setting.overdue),
```

Comment:
I might be misunderstanding this, but don't we know the exact types of the form groups & controls here?

Reply:
That's just the angular migration running and transforming it automatically. I haven't looked into detail on how we can type it.